runners/spark/src/main/java/org/apache/beam/runners/spark/translation/GroupCombineFunctions.java

```diff
@@ -52,13 +50,11 @@
             .map(WindowingHelpers.unwindowFunction())
             .mapToPair(TranslationUtils.toPairFunction())
             .mapToPair(CoderHelpers.toByteFunction(keyCoder, wvCoder));
-    // use a default parallelism HashPartitioner.
-    Partitioner partitioner = new HashPartitioner(rdd.rdd().sparkContext().defaultParallelism());
 
     // using mapPartitions allows to preserve the partitioner
     // and avoid unnecessary shuffle downstream.
     return pairRDD
-        .groupByKey(partitioner)
+        .groupByKey()
```

Comment:
There is a previous issue on double shuffling that was solved by explicitly passing the partitioner [BEAM-1815](https://jira.apache.org/jira/browse/BEAM-1815). I am afraid this change has the side effect of bringing this issue back.

Reply:
To me it appears the double shuffling was solved by changing map and mapToPair to mapPartitions and mapPartitionsToPair, respectively. I don't think the explicit declaration of the HashPartitioner has any effect.

Reply:
If I am incorrect, how would we know whether to use the defaultParallelism or some other value? I don't think it would be appropriate to force a SourceRDD, that may have had hundreds of partitions, into the defaultParallelism number of partitions, which may be quite small, as this may result in too much data being in each partition.